payments/src/main/java/it/gov/pagopa/payments/service/PaymentsService.java

```diff
@@ -114,10 +117,16 @@ public PaymentsResult<ReceiptEntity> getOrganizationReceipts(@Positive Integer l
 					.getTableReference(receiptsTable);
 
 			String filter = TableQuery.generateFilterCondition(PARTITION_KEY_FIELD, TableQuery.QueryComparisons.EQUAL, organizationFiscalCode);
+			
 			if (null != debtor) {
 				String debtorFilter       = TableQuery.generateFilterCondition(DEBTOR_FIELD, TableQuery.QueryComparisons.EQUAL, debtor);
 				filter = TableQuery.combineFilters(filter, Operators.AND, debtorFilter);
 			}
+			
+			if (null != service) {
+				String serviceFilter = PaymentsService.getStartsWithFilter(ROW_KEY_FIELD, service);
```

Comment:
rename `serviceFilter` with `segretationCodeFilter` to rember us that is a special case.
in the near future ( i hope ) we should service in more appropriately

Reply:
Are you sure? In my opinion the implementation should be aseptic regarding the momentary meaning of a data that is passed in input